src/operations/profiling_level.ts

```diff
@@ -1,39 +1,42 @@
 import type { Db } from '../db';
-import { MongoRuntimeError } from '../error';
+import { MongoUnexpectedServerResponseError } from '../error';
 import type { Server } from '../sdam/server';
 import type { ClientSession } from '../sessions';
-import type { Callback } from '../utils';
-import { CommandCallbackOperation, type CommandOperationOptions } from './command';
+import { type Callback } from '../utils';
+import { CommandOperation, type CommandOperationOptions } from './command';
 
 /** @public */
 export type ProfilingLevelOptions = CommandOperationOptions;
 
 /** @internal */
-export class ProfilingLevelOperation extends CommandCallbackOperation<string> {
+export class ProfilingLevelOperation extends CommandOperation<string> {
   override options: ProfilingLevelOptions;
 
   constructor(db: Db, options: ProfilingLevelOptions) {
     super(db, options);
     this.options = options;
   }
 
-  override executeCallback(
-    server: Server,
-    session: ClientSession | undefined,
-    callback: Callback<string>
+  override async execute(server: Server, session: ClientSession | undefined): Promise<string> {
+    const doc = await super.executeCommand(server, session, { profile: -1 });
+    if (doc.ok === 1) {
+      const was = doc.was;
+      if (was === 0) return 'off';
+      if (was === 1) return 'slow_only';
+      if (was === 2) return 'all';
+      // TODO(NODE-3483)
+      throw new MongoUnexpectedServerResponseError(`Illegal profiling level value ${was}`);
```

Comment:
The TODOs can be removed since they are addressed here, and I would mention this change in the PR description.

Reply:
removed